### ISSUE TITLE
Write Permissions Cannot Create Client

### ISSUE THREAD

**Issue by thejeff77** [triaxtec/openapitools-generator-action#21]
Seeing this stack trace continually with any openapi spec.. seems to not have permission to write or something.

```
[main] INFO  o.o.codegen.TemplateManager - writing file /github/workspace/kotlin-client/src/main/kotlin/com/company/project/model/SomeType.kt
Exception in thread "main" java.lang.RuntimeException: Could not generate model 'AccountType'
	at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:569)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:946)
	at org.openapitools.codegen.cmd.Generate.execute(Generate.java:487)
	at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Caused by: java.nio.file.NoSuchFileException: /github/workspace/kotlin-client/src/main/kotlin/com/company/project/model/SomeType.kt
	at java.base/sun.nio.fs.UnixException.translateToIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(Unknown Source)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(Unknown Source)
	at java.base/java.nio.file.Files.newOutputStream(Unknown Source)
	at java.base/java.nio.file.Files.write(Unknown Source)
	at org.openapitools.codegen.TemplateManager.writeToFileRaw(TemplateManager.java:255)
	at org.openapitools.codegen.TemplateManager.writeToFile(TemplateManager.java:237)
	at org.openapitools.codegen.TemplateManager.writeToFile(TemplateManager.java:198)
	at org.openapitools.codegen.TemplateManager.write(TemplateManager.java:1[64](https://github.com/company/project-service/actions/runs/5203732454/jobs/9387007786#step:4:66))
	at org.openapitools.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:1092)
	at org.openapitools.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:10[79](https://github.com/company/project/actions/runs/5203732454/jobs/9387007786#step:4:81))
	at org.openapitools.codegen.DefaultGenerator.generateModel(DefaultGenerator.java:408)
	at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:560)
```

**Comment by thejeff77** [comment]
Someone smart helped me fix this. I changed the user from `1001` to `1000`

`runner:x:1000:1000:,,,:/home/runner:/bin/bash`. 

It's mounted as a user that doesn't exist in our runner

hard coding permissions is a poor practice, and this would be an intermittent failure caused by action configuration.

**Comment by dbanty** [comment]
We set the UID for a reason, but I don't remember what. @emann do you remember why we needed that?

**Comment by alexef** [comment]
hitting the same issue. should the user be an input?

**Comment by thejeff77** [comment]
Yeah probably. To get it to work I had to fork it and use user 1000. 

Feel free to use my fork.. it worked for me at least.

**Comment by thejeff77** [comment]
@alexef don't forget to update the docs for your PR

**Comment by dbanty** [comment]
Should be fixed in 1.4.1, thanks!
